### PR TITLE
Expand called before logits were converted into LogSemiring Matrix

### PR DESCRIPTION
**Pull request by MartinKocour**

### CODE REVIEW

**Comment by MartinKocour** [comment]
For some reason I like to complicate things, even when it is not necessary :)

**Comment by lucasondel** [comment]
Everybody does :)

**Review by lucasondel — APPROVED** [review]
